common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -12,52 +12,33 @@
 
 public class StringUtils {
   /**
-   * Unquote any string with mark specified.
-   * @param text string
-   * @param mark quotation mark
-   * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
-   *     removed
-   */
-  public static String unquote(String text, String mark) {
-    if (isQuoted(text, mark)) {
-      return text.substring(mark.length(), text.length() - mark.length());
-    }
-    return text;
-  }
-
-  /**
-   * Unquote Identifier which has " or ' or ` as mark.
+   * Unquote Identifier which has " or ' as mark.
    * Strings quoted by ' or " with two of these quotes appearing next to each other in the quote
    * acts as an escape
    * Example: 'Test''s' will result in 'Test's', similar with those single quotes being replaced
-   * with double.
+   * with double quote.
+   * Supports escaping quotes (single/double) and escape characters using the `\` characters.
    * @param text string
-   * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
+   * @return An unquoted string whose outer pair of (single/double) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-
     if (text.length() < 2) {
       return text;
     }
 
-    char enclosingQuote;
+    char enclosingQuote = 0;
     char firstChar = text.charAt(0);
```

Comment:
suggest removing `firstChar` and just using `enclosingQuote`

Reply:
I don't like setting enclosing to something like that in the case where the starting quote != ending quote. Could cause a problem later.